torch/csrc/deploy/interpreter/interpreter_impl.h

```diff
@@ -1,26 +1,98 @@
 #pragma once
+// multi-python abstract code
 #include <ATen/ATen.h>
+#include <ATen/core/ivalue.h>
+#include <caffe2/serialize/inline_container.h>
 
-// NOTE- if adding new interface functions,
-// update interpreter.cpp initialize_interface.
-size_t load_model(const char* model_file, bool hermetic=false);
-at::Tensor forward_model(size_t model_id, at::Tensor const & input);
-void run_some_python(const char* code);
-void startup();
-void teardown();
-void run_python_file(const char* code);
+namespace torch {
 
+struct InterpreterSessionImpl;
 
-#define FOREACH_INTERFACE_FUNCTION(_) \
-  _(load_model)                       \
-  _(forward_model)                    \
-  _(run_some_python)                  \
-  _(startup)                          \
-  _(teardown)                         \
-  _(run_python_file)
+struct PickledObject {
+  std::string data_;
+  std::vector<at::Storage> storages_;
+  // types for the storages, required to
+  // reconstruct correct Python storages
+  std::vector<at::ScalarType> types_;
+  std::shared_ptr<caffe2::serialize::PyTorchStreamReader> container_file_;
+};
+
+struct PythonObject {
```

Comment:
some comment on design of PythonObject would be helpful, i.e. why do we need this wrapping